### PR TITLE
Feature/rpc error handling

### PR DESCRIPTION
**Pull request by joniba**
this is a small patch to protect consumers from unexpected errors in request-reply queues. I changed some of your timeouts to decrease the execution time of the tests, which was very long.

### CODE REVIEW

File: test/request-reply.spec.js [hunk 1]
```diff
@@ -15,85 +18,71 @@ describe('request-reply should: ', function () {
...
-            url,
-            queues: {
-                testReplyTo: {
-                    name: 'test-reply-to',
-                    requestReply: true
-                }
-            }
+        let serverReceived;
+        await server.consume(async (data, props) => {
+            await Promise.delay(50);
```
Comment (nati-maskens) [inline test/request-reply.spec.js]: Speaking about time saving in tests, as I understood that with RPC the publish is literally awaiting on the consume, why this delay is necessary at all?

**Review by nati-maskens — CHANGES_REQUESTED** [review]
Great, tests are better so indeed!
- Please add release notes :grin:

File: test/request-reply.spec.js (hunk 1 above)
Comment (joniba) [reply test/request-reply.spec.js]: I don't know why... I removed it

**Comment by joniba** [comment]
all fixed, merging

**Pull request closed without merging**